rosserial_server/include/rosserial_server/debug_server.h

```diff
@@ -53,18 +55,66 @@ using boost::asio::ip::tcp;
 class DebugServer
 {
 public:
-  DebugServer(boost::asio::io_service& io_service, short port = 11412)
+  DebugServer(boost::asio::io_service& io_service, unsigned short port)
     : io_service_(io_service),
       acceptor_(io_service, tcp::endpoint(tcp::v4(), port))
   {
     start_accept();
   }
 
-  void log(const std::vector<uint8_t>& buffer, char incoming_or_outgoing, const std::string& topic = {}) {
+  DebugServer(boost::asio::io_service& io_service)
+    : io_service_(io_service),
+      acceptor_(io_service)
+  {}
+
+  ~DebugServer()
+  {
+    closeLogfile();
+  }
+
+  void listen(unsigned short port)
+  {
+    if (acceptor_.is_open()) acceptor_.cancel();
+    acceptor_ = tcp::acceptor(acceptor_.get_executor(), tcp::endpoint(tcp::v4(), port));
+  }
```

Comment:
missing a `start_accept();` ?

Reply:
Yes, apparently I never tested this version anymore after the latest changes, only the version where `start_accept()` is already called from the constructor of `DebugServer`. Good catch!